client/src/accounts/authentication/login.hook.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { describe, expect, it } from 'vitest';
+import { rest, server } from '~/mocks/server';
+import { act, renderHook, waitFor } from '~/test/utils';
+import { useLogin } from './login.hook';
+const ENDPOINT = '*/api/accounts/';
+
+interface Result {
+  isSuccess: boolean;
+  isError: boolean;
+  data: Record<string, unknown>;
+  mutate: (form: Record<string, unknown>) => void;
+  error: Record<string, unknown>;
+  current: Record<string, unknown>;
+}
+
+describe('useLoginHook', () => {
+  it('should throw an error for an unknown user', async () => {
+    const loginData = { email: 'bob@example.com', password: 'otherpassword', accessToken: 'foobar' };
+    const { result } = renderHook<Result, unknown>(() => useLogin());
+    server.use(rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(400), ctx.json({ message: 'User not found' }))));
+    act(() => {
+      result.current.mutate(loginData);
+    });
+    await waitFor(() => {
+      expect(result.current.isError).toBe(true);
+      expect(result.current.error).toEqual({ message: 'User not found' });
+    });
+  });
+  it('should throw an error for an known user with wrong password', async () => {
+    const loginData = { email: 'john@example.com', password: 'otherpassword', accessToken: 'foobar' };
+    const { result } = renderHook(() => useLogin());
+    server.use(
+      rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(400), ctx.json({ message: 'Incorrect password' }))),
+    );
+    act(() => {
+      result.current.mutate(loginData);
+    });
+    await waitFor(() => {
```

Comment:
This doesn't really follow the pattern I stated in slack 2 days ago. The maintainers of react-query, at least I think it was them, suggest awaiting the `result.current.isSuccess`, then after that, the expect lines can just be synchronous and nothing needs to be or should be returned from a test or an await block. If we follow a common pattern, then everyone's life is made easier.